src/main/java/ca/on/oicr/gsi/dimsum/service/CaseService.java

```diff
@@ -404,6 +334,23 @@ public TableData<ProjectSummaryRow> getProjectSummaryRows(String name) {
     return data;
   }
 
+  public TableData<ProjectSummaryRow> getFilteredProjectSummaryRows(String projectName,
+      List<CaseFilter> filters) {
+    List<Case> cases = filterCases(caseData.getCases(), filters).toList();
+    Map<String, ProjectSummary> projectSummariesByName =
+        CaseLoader.calculateProjectSummaries(cases);
+    ProjectSummary projectSummary = projectSummariesByName.get(projectName);
```

Comment:
This works for filters that only apply at the case level, but is not working correctly for filters that should separate tests. For example: when using the "Test" filter, this returns numbers that include all tests for each case where one of the tests matches the filter. It should instead only count the matching test.

Reply:
Should we then only show project summary at the test level when "Test" filters are applied (ex. extraction, library prep., etc)?

Reply:
It should still count all gates. Example: say you have a case with 2 tests. Receipt is completed for the case, and extraction is completed for both tests.

* with no filter, this should show as having 2 completed receipt, and 2 completed extraction
* if you filter to match one of the tests, it should show 1 completed receipt and 1 completed extraction

This will make your filtering and calculation methods more complex. You'll need to filter the cases, keep that information, and also filter the tests, then calculate based only on those tests, but with the case-level data included too. Might want to create some structure like `List<Pair<Test, Case>>` or `Map<Case, <List<Test>>` when filtering

Reply:
What is the difference between creating the new structure and using the `filterTestTableViews`? Doesn't `TestTableViews` already includes case information?